environs/config/config.go

```diff
@@ -477,7 +477,7 @@ func New(withDefaults Defaulting, attrs map[string]any) (*Config, error) {
 	}
 
 	// no old config to compare against
-	if err := Validate(context.TODO(), c, nil); err != nil {
+	if err := Validate(context.Background(), c, nil); err != nil {
```

Comment:
Is this OK though? looks like we should maybe have a context passed through to `New()`. In that case we keep the .TODO() right?
Having .Background() in non-test code isn't ideal.

Reply:
`TODO` is for "fixme". `Background` is for when we don't plan on doing something about it right now. Not every method or function should take a `context.Context`, that's ok. We shouldn't let a `context.Context` pollute every function or method, we should draw a line somewhere.

Reply:
I understand, my only point here is that having a `.TODO()` simply tells you that you had to "draw the line" there, even if it's not perfect. This is a compromise that you might want to revisit in the future (or not).
`.Background()` on the other hand, is a decision on using the background (or default) context in that function, without any statement about the "compromise" you had to do. Also, it's not easily searchable if we want to address in the future.

Reply:
Looking at validate function I didn't think there was anything we'd want to change right now. It's up to people to fix that later.